src/Sarif.Multitool/Rules/SARIF2005.ProvideHelpfulToolInformation.cs

```diff
@@ -0,0 +1,93 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using Microsoft.Json.Pointer;
+using System.Collections.Generic;
+using System.Text.RegularExpressions;
+
+namespace Microsoft.CodeAnalysis.Sarif.Multitool.Rules
+{
+    public class ProvideHelpfulToolInformation : SarifValidationSkimmerBase
+    {
+        public override MultiformatMessageString FullDescription => new MultiformatMessageString
+        {
+            Text = RuleResources.SARIF2005_ProvideHelpfulToolInformation_FullDescription_Text
+        };
+
+        public override FailureLevel DefaultLevel => FailureLevel.Warning;
+
+        public override string Id => RuleId.ProvideHelpfulToolInformation;
+
+        protected override IEnumerable<string> MessageResourceNames => new string[]
+        {
+            nameof(RuleResources.SARIF2005_ProvideHelpfulToolInformation_Warning_ProvideToolVersion_Text),
+            nameof(RuleResources.SARIF2005_ProvideHelpfulToolInformation_Warning_ProvideConciseToolName_Text),
+            nameof(RuleResources.SARIF2005_ProvideHelpfulToolInformation_Warning_UseNumericToolVersions_Text)
+        };
+
+        protected override void Analyze(Tool tool, string toolPointer)
+        {
+            if (tool == null)
+            {
+                return;
+            }
+
+            if (tool.Driver != null)
+            {
+                AnalyzeToolDriver(tool.Driver, toolPointer);
+            }
+        }
+
+        private void AnalyzeToolDriver(ToolComponent toolComponent, string toolPointer)
+        {
+            // ProvideConciseToolName: Ensure that tool.driver.name isn't more than 3 words long
+            if (!string.IsNullOrEmpty(toolComponent.Name))
+            {
+                const int maxWordCount = 3;
+                int wordCount = toolComponent.Name.Split(' ').Length;
```

Comment:
>Split [](start = 51, length = 5)

Use `StringSplitOptions.RemoveEmptyEntries` in case there is more than one space between words.

Reply:
for net461, we need to do a cast to (char).


---
In reply to: [444573658](https://github.com/microsoft/sarif-sdk/pull/1926#discussion_r444573658) [](ancestors = 444573658)

Reply:
Yikes, really??? That's can't be right, let me look. I think you might be using the wrong overload.

---
In reply to: [444575924](https://github.com/microsoft/sarif-sdk/pull/1926#discussion_r444575924) [](ancestors = 444575924,444573658)

Reply:
at least, net461 didnt like it...

---
In reply to: [444577532](https://github.com/microsoft/sarif-sdk/pull/1926#discussion_r444577532) [](ancestors = 444577532,444575924,444573658)

Reply:
Yes, this is wrong. It should be `Split(new char[] { ' ' }, StringSplitOptions.RemoveEmptyEntries)`. Take a look at the list of overloads of `String.Split` to see why.

---
In reply to: [444577532](https://github.com/microsoft/sarif-sdk/pull/1926#discussion_r444577532) [](ancestors = 444577532,444575924,444573658)